submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/EmptyAccNoApiTest.kt

```diff
@@ -0,0 +1,68 @@
+package ac.uk.ebi.biostd.itest.test.submission.submit
+
+import ac.uk.ebi.biostd.client.integration.commons.SubmissionFormat
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import ac.uk.ebi.biostd.common.config.PersistenceConfig
+import ac.uk.ebi.biostd.itest.common.BaseIntegrationTest
+import ac.uk.ebi.biostd.itest.entities.SuperUser
+import ac.uk.ebi.biostd.persistence.model.Counter
+import ac.uk.ebi.biostd.persistence.model.Sequence
+import ac.uk.ebi.biostd.persistence.repositories.SequenceDataRepository
+import ac.uk.ebi.biostd.persistence.service.SubmissionRepository
+import ebi.ac.uk.asserts.assertThat
+import ebi.ac.uk.dsl.line
+import ebi.ac.uk.dsl.submission
+import ebi.ac.uk.dsl.tsv
+import ebi.ac.uk.model.extensions.title
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import io.github.glytching.junit.extension.folder.TemporaryFolderExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+@ExtendWith(TemporaryFolderExtension::class)
+internal class EmptyAccNoApiTest(tempFolder: TemporaryFolder) : BaseIntegrationTest(tempFolder) {
+    @Nested
+    @Import(PersistenceConfig::class)
+    @ExtendWith(SpringExtension::class)
+    @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+    @DirtiesContext
+    inner class EmptyAccNoApiTest(
+        @Autowired val submissionRepository: SubmissionRepository,
+        @Autowired val sequenceRepository: SequenceDataRepository
+    ) {
+        @LocalServerPort
+        private var serverPort: Int = 0
+
+        private lateinit var webClient: BioWebClient
+
+        @BeforeAll
+        fun init() {
+            webClient = getWebClient(serverPort, SuperUser)
+            sequenceRepository.save(Sequence("S-BSST", "").apply { counter = Counter("S-BSST") })
```

Comment:
it looks handy to create a sequence constructor that creates the counter too.

Reply:
Done